agent/taskresource/credentialspec/credentialspec_linux.go

```diff
@@ -0,0 +1,455 @@
+	// Example item := arn:aws:ssm:us-east-1:XXXXXXXXXXXXX:parameter/x/y/c:container-sql
+	// This stores the map of a credential spec to corresponding container name
+	credentialSpecContainerMap map[string]string
+	//	This stores credspec contents associated to all the containers of the task
+	credentialsFetcherRequest []string
+	// lock is used for fields that are accessed and updated concurrently
+	lock sync.RWMutex
+}
+
+// ServiceAccountInfo contains account info associated to a credentialspec
+type ServiceAccountInfo struct {
+	serviceAccountName string
+	domainName         string
+}
+
+// CredentialSpec object schema
+type CredentialSpec struct {
+	CmsPlugins       []string `json:"CmsPlugins"`
+	DomainJoinConfig struct {
+		Sid                string `json:"Sid"`
+		MachineAccountName string `json:"MachineAccountName"`
+		GUID               string `json:"Guid"`
+		DNSTreeName        string `json:"DnsTreeName"`
+		DNSName            string `json:"DnsName"`
+		NetBiosName        string `json:"NetBiosName"`
+	} `json:"DomainJoinConfig"`
+	ActiveDirectoryConfig struct {
+		GroupManagedServiceAccounts []struct {
+			Name  string `json:"Name"`
+			Scope string `json:"Scope"`
+		} `json:"GroupManagedServiceAccounts"`
+	} `json:"ActiveDirectoryConfig"`
+}
+
+// NewCredentialSpecResource creates a new CredentialSpecResource object
+func NewCredentialSpecResource(taskARN, region string,
+	executionCredentialsID string,
+	credentialsManager credentials.Manager,
+	ssmClientCreator ssmfactory.SSMClientCreator,
+	asmClientCreator asmfactory.ClientCreator,
+	s3ClientCreator s3factory.S3ClientCreator,
+	credentialSpecContainerMap map[string]string) (*CredentialSpecResource, error) {
+	s := &CredentialSpecResource{
+		taskARN:                    taskARN,
+		region:                     region,
+		credentialsManager:         credentialsManager,
+		executionCredentialsID:     executionCredentialsID,
+		ssmClientCreator:           ssmClientCreator,
+		asmClientCreator:           asmClientCreator,
+		s3ClientCreator:            s3ClientCreator,
+		CredSpecMap:                make(map[string]string),
+		credentialSpecContainerMap: credentialSpecContainerMap,
+		ServiceAccountInfoMap:      make(map[string]ServiceAccountInfo),
+	}
+
+	s.initStatusToTransition()
+	return s, nil
+}
+
+func (cs *CredentialSpecResource) Initialize(resourceFields *taskresource.ResourceFields,
+	_ status.TaskStatus,
+	_ status.TaskStatus) {
+
+	cs.credentialsManager = resourceFields.CredentialsManager
+	cs.ssmClientCreator = resourceFields.SSMClientCreator
+	cs.asmClientCreator = resourceFields.ASMClientCreator
+	cs.s3ClientCreator = resourceFields.S3ClientCreator
+	cs.initStatusToTransition()
+}
+
+// Create is used to retrieve credentialspec resources for a given task
+func (cs *CredentialSpecResource) Create() error {
+	var iamCredentials credentials.IAMRoleCredentials
+
+	executionCredentials, ok := cs.credentialsManager.GetTaskCredentials(cs.getExecutionCredentialsID())
+	if ok {
```

Comment:
If not ok, would nil iamCredentials break line 193, 196, and 199?

Reply:
it won't break , if the iam credentials are not able to fetched  sdk will return false